testsuite/MDAnalysisTests/analysis/test_psa.py

```diff
@@ -176,152 +174,163 @@ def test_dist_mat_to_vec_func_bad_integers(self):
-        del self.random_columns_2
-        del self.path_1
-        del self.path_2
-
-    def test_symmetry(self):
+        random_columns_2[0, 0] = np.cos(random_columns_2[0, 0]) * 3.3
+        random_columns_2[0, 1] = np.sin(random_columns_2[0, 1]) * 3.3
+        return random_columns_2
+
+    def test_symmetry(self, path_1, path_2, h):
         '''Ensure that the undirected (symmetric)
         Hausdorff distance is actually symmetric
         for a given Hausdorff metric, h.'''
-        forward = self.h(self.path_1, self.path_2)
-        reverse = self.h(self.path_2, self.path_1)
+        forward = h(path_1, path_2)
+        reverse = h(path_2, path_1)
         # lower precision on 32bit
         assert_almost_equal(forward, reverse, decimal=15)
 
-    def test_hausdorff_value(self):
+    def test_hausdorff_value(self, path_1, path_2, h, expected):
         '''Test that the undirected Hausdorff
         distance matches expected value for
         the simple case here.'''
-        actual = self.h(self.path_1, self.path_2)
+        actual = h(path_1, path_2)
         # unless I pin down the random generator
         # seems unstable to use decimal > 2
-        assert_almost_equal(actual, self.expected,
-                            decimal=2)
+        assert_almost_equal(actual, expected, decimal=2)
+
 
 class TestHausdorffSymmetric(_BaseHausdorffDistance):
     '''Tests for conventional and symmetric (undirected)
     Hausdorff distance between point sets in 3D.'''
 
-    __test__ = True
+    # expected = 2.3
+
+    @pytest.fixture()
+    def expected(self):
+        return 2.3
+
+    # TODO: Fix this hack! without the fixture the test says I passed 3
+    # params instead of 2 (I think it sends self as a parameter too.)
+    @pytest.fixture()
+    def h(self):
+        return PSA.hausdorff
 
-    def setUp(self):
-        super(TestHausdorffSymmetric, self).setUp()
-        self.h = PSA.hausdorff
-        # radii differ by ~ 2.3 for outlier
-        self.expected = 2.3
 
 class TestWeightedAvgHausdorffSymmetric(_BaseHausdorffDistance):
     '''Tests for weighted average and symmetric (undirected)
     Hausdorff distance between point sets in 3D.'''
 
-    __test__ = True
+    @pytest.fixture()
+    def expected(self, path_1, path_2):
+        distance_matrix = scipy.spatial.distance.cdist(path_1, path_2)
+        return (np.mean(np.amin(distance_matrix, axis=0)) +
+                np.mean(np.amin(distance_matrix, axis=1))) / 2.0
 
-    def setUp(self):
-        super(TestWeightedAvgHausdorffSymmetric, self).setUp()
-        self.h = PSA.hausdorff_wavg
-        self.distance_matrix = scipy.spatial.distance.cdist(self.path_1,
-                                                            self.path_2)
-        self.expected = (np.mean(np.amin(self.distance_matrix, axis=0)) +
-                         np.mean(np.amin(self.distance_matrix, axis = 1))) / 2.
+    # TODO: Fix this hack! without the fixture the test says I passed 3
```

Comment:
I did try that, but when I make it a class attribute, the test fails saying that `h` expected 2 parameters but got 3.